src/CollectorGuess.cpp

```diff
@@ -48,6 +48,17 @@ bool isNumber(const std::string& x, LocaleInfo* pLocale) {
   return ok && begin == x.begin() && end == x.end();
 }
 
+bool isInteger(const std::string& x, LocaleInfo* pLocale) {
```

Comment:
integer guessing was explicitly removed in https://github.com/tidyverse/readr/pull/655/files, you should not be adding it back here.